code/espurna/config/dependencies.h

```diff
@@ -5,6 +5,10 @@
 // Configuration settings are in the general.h file
 //------------------------------------------------------------------------------
 
+#if !defined(ARDUINO_ESP8266_RELEASE_2_5_0) && !defined(ARDUINO_ESP8266_RELEASE_2_5_1) && !defined(ARDUINO_ESP8266_RELEASE_2_5_2)
+#define USING_AXTLS // do not use BearSSL on core < 2.5.0
+#endif
+
```

Comment:
We need a way to check this in a better way - maybe ``ESP.getCoreVersion()``?

Reply:
That way bearssl is also off with git versions of the Core.
2.5.0 and up introduced https://github.com/esp8266/Arduino/blob/master/cores/esp8266/core_esp8266_version.h
tag with proper comparison, but we again are hit by multi-Core support :/

In theory, one can depend on some behaviour / implementation instead. Kind of dangerous though, if that changes. For example, see:
https://github.com/xoseperez/espurna/compare/dev...mcspr:utils/fragmentation

Reply:
That way bearssl is also off with git versions of the Core.
2.5.0 and up introduced https://github.com/esp8266/Arduino/blob/master/cores/esp8266/core_esp8266_version.h
tag with proper comparison, but we again are hit by multi-Core support :/

In theory, one can depend on some behaviour / implementation instead. Kind of dangerous though, if it changes. For example, see:
https://github.com/xoseperez/espurna/compare/dev...mcspr:utils/fragmentation

Reply:
Maybe we can use ``ARDUINO_ESP8266_GIT_VER`` instead? The type_traits stuff looks pretty rough (not to mention badly readable).

Reply:
ARDUINO_ESP8266_GIT_VER has the same problem as RELEASE defines. BTW I think we are carried away. Does this work? Shouldn't it be global flag instead? IMO documentation would solve this better. If we sure this breaks horribly, static_assert(condition, "message") can also do this.

Regarding other branch, you'd say that this is a clearer intent?
https://github.com/xoseperez/espurna/blob/9ceca44770cc7772afc020ce7bad2d4cf321eadb/code/espurna/config/prototypes.h#L55-L66

https://en.cppreference.com/w/cpp/language/decltype
https://en.cppreference.com/w/cpp/types/enable_if
It is just an overload detection. We force compiler to pick specific 'detect()' method and store bool based on the one it found (either internal or external return type)
From my limited understanding, to make this shorter 'getHeapStats' function would need different return types.